src/main.rs

```diff
@@ -1,73 +1,83 @@
+        .fold(0, |correct_answers, (i, line)| {
+            // clear_console(&mut stdout);
+            let (line, hidden_word) = hide_word(line);
+            stdout.queue(Print(line.blue())).unwrap();
+            stdout.queue(Print("> ")).unwrap();
+            stdout.flush().unwrap();
+            let input = stdin.next().unwrap().unwrap();
+            let input = input.trim();
+            let is_valid = input.eq_ignore_ascii_case(&hidden_word);
+            let (correct_answers, foreground_color, message) = if is_valid {
+                (
+                    correct_answers + 1,
+                    Color::Green,
+                    format!("Correct answer!"),
+                )
             } else {
-                bad_answers += 1;
-                stdout.queue(SetForegroundColor(Color::Red)).unwrap();
-                print!("Wrong! ({hidden_word})");
-            }
-        }
-        let all_answers = good_answers + bad_answers;
-        println!(" {good_answers}/{all_answers}");
-        stdout.queue(ResetColor).unwrap();
-        println!("{}", "-".repeat(TEXT_WIDTH as usize));
-    }
+                (
+                    correct_answers,
+                    Color::Red,
+                    format!("Wrong answer! Correct = \"{hidden_word}\""),
+                )
+            };
+            stdout.queue(SetForegroundColor(foreground_color)).unwrap();
+            stdout.queue(Print(format!("{message} "))).unwrap();
+            stdout
+                .queue(Print(format!("{correct_answers}/{}\n", i + 1)))
+                .unwrap();
+            stdout.queue(ResetColor).unwrap();
+            stdout
+                .queue(Print("-".repeat(TEXT_WIDTH as usize)))
+                .unwrap();
+            stdout.queue(Print("\n\n")).unwrap();
+            stdout.flush().unwrap();
+            correct_answers
+        });
 }
 
-fn hide_word(line: &str) -> (String, String) {
-    lazy_static! {
-        static ref WORD_RE: Regex = Regex::new(r"(?P<start>.*?)(?P<word>\w+)(?P<end>.*)").unwrap();
-    }
-    let mut rand = rand::thread_rng();
-    let words = line.split_whitespace();
-    let word_count = words.clone().count();
-    let mut words: Vec<&str> = words.collect();
+// here we're using AsRef<str> instead of &str or String, because AsRef<str> can accept both of these.
+fn hide_word(line: impl AsRef<str>) -> (String, String) {
+    let line = line.as_ref();
+    let words = line.split_whitespace().collect::<Vec<_>>();
+    let n = rand::thread_rng().gen_range(0..words.len());
+    let hidden_word = words.get(n).unwrap().to_string();
+    // eventually replace it with some constant if you want to make it harder for user to recognize the hidden word
+    let censored_word = "_".repeat(hidden_word.len());
 
-    let index = rand.gen_range(0..word_count);
-    let word = words[index];
-    let captures = WORD_RE.captures(word).unwrap();
-    let hidden_word: String = captures.name("word").unwrap().as_str().to_string();
```

Comment:
This regex removes punctuation marks from the word (like `,.?!`) and you shouldn't remove it